modules/samples/watertank/component/api/src/main/java/org/homeautomation/watertank/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+                return Response.status(Response.Status.NOT_ACCEPTABLE.getStatusCode()).build();
+            }
+        } catch (DeviceManagementException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        } catch (DeviceAccessAuthorizationException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        }
+    }
+
+    /**
+     * To get device information
+     *
+     * @param deviceId unique identifier for given device type instance
+     * @return
+     */
+    @Path("/device/{deviceId}")
+    @GET
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getDevice(@PathParam("deviceId") String deviceId) {
+        try {
+            DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+            deviceIdentifier.setId(deviceId);
+            deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+            if (!APIUtil.getDeviceAccessAuthorizationService().isUserAuthorized(deviceIdentifier)) {
+                return Response.status(Response.Status.UNAUTHORIZED.getStatusCode()).build();
+            }
+            Device device = APIUtil.getDeviceManagementService().getDevice(deviceIdentifier);
+            return Response.ok().entity(device).build();
+        } catch (DeviceManagementException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        } catch (DeviceAccessAuthorizationException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        }
+    }
+
+    /**
+     * Get all device type instance which belongs to user
+     *
+     * @return Array of devices which includes device's information
+     */
+    @Path("/devices")
+    @GET
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAllDevices() {
+        try {
+            List<Device> userDevices =
+                    APIUtil.getDeviceManagementService().getDevicesOfUser(APIUtil.getAuthenticatedUser());
+            ArrayList<Device> userDevicesforwatertank = new ArrayList<>();
+            for (Device device : userDevices) {
+                if (device.getType().equals(DeviceTypeConstants.DEVICE_TYPE) &&
+                    device.getEnrolmentInfo().getStatus().equals(EnrolmentInfo.Status.ACTIVE)) {
+                    userDevicesforwatertank.add(device);
+                }
+            }
+            Device[] devices = userDevicesforwatertank.toArray(new Device[]{});
+            return Response.ok().entity(devices).build();
+        } catch (DeviceManagementException e) {
+            log.error(e.getErrorMessage(), e);
```

Comment:
repeating the same error message (i.e. e.getErrorMessage()) here, which can be found in the stack trace as well, is not a good practice.